annotation-file-utilities/src/main/java/org/checkerframework/afu/annotator/find/IsSigMethodCriterion.java

```diff
@@ -56,55 +62,23 @@ public Context(String packageName, List<String> imports) {
   /**
    * Creates a new IsSigMethodCriterion.
    *
-   * @param fullSignature the full JVML signature
+   * @param fullSignature the full JVML signature (that is, a method descriptor)
    */
-  public IsSigMethodCriterion(String fullSignature) {
-    this.signature = fullSignature.substring(0, fullSignature.indexOf(")") + 1);
+  public IsSigMethodCriterion(@MethodDescriptor String fullSignature) {
+    this.signatureWithoutReturnType = fullSignature.substring(0, fullSignature.indexOf(")") + 1);
     this.simpleMethodName = fullSignature.substring(0, fullSignature.indexOf("("));
-    //    this.fullyQualifiedParams = new ArrayList<String>();
-    //    for (String s : fullSignature.substring(
-    //        fullSignature.indexOf("(") + 1, fullSignature.indexOf(")")).split(",")) {
-    //      if (s.length() > 0) {
-    //        fullyQualifiedParams.add(s);
-    //      }
-    //    }
-    this.fullyQualifiedParams = new ArrayList<>();
     try {
-      parseParams(
-          fullSignature.substring(fullSignature.indexOf("(") + 1, fullSignature.indexOf(")")));
+      String jvmlArgs =
+          fullSignature.substring(fullSignature.indexOf("("), fullSignature.indexOf(")") + 1);
+      this.fullyQualifiedParams =
+          CollectionsPlume.mapList(
+              Signatures::fieldDescriptorToBinaryName, Signatures.splitJvmArglist(jvmlArgs));
```

Comment:
`Signatures.splitJvmArglist` and `Signatures.methodDescriptorToReturnType` only exist in reflection-util versions that do no longer work with Java 8.
We could work around this, but for AFU I'm fine with ending Java 8 support.